### PR TITLE
Fixes #588: Refactored review to revision

### LINKED ISSUE #588: Review should be updated to Revision
As per new mocks, Topic Review Tab has been updated to Topic Revision Tab,

Screenshot 2020-01-03 at 12 11 29 PM

So, all files related to that needs to be updated like package name, file names, test cases names, xml files names, variable names, strings.xml, drawables, etc.

In short refactor `Review` with `Revision`.

### PR DESCRIPTION
**Pull request by abhinavraj23**
## Explanation
Refactored Review to Revision

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package org.oppia.app.topic
 
 /** Listener for when an [TopicActivity] should route to a [ReviewCardFragment]. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt]: Update this comment also.

File: app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-interface RouteToReviewCardListener {
+interface RouteToRevisionCardListener {
   fun routeToReviewCard(topicId: String, subtopicId: String)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt]: Update this function name also.

File: app/src/main/java/org/oppia/app/topic/TopicActivity.kt [hunk 3]
```diff
@@ -45,7 +45,7 @@ class TopicActivity : InjectableAppCompatActivity(), RouteToQuestionPlayerListen
   }
 
   override fun routeToReviewCard(topicId: String, subtopicId: String) {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/TopicActivity.kt]: Ditto

File: app/src/main/java/org/oppia/app/topic/revision/RevisionSubtopicSelector.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.oppia.app.topic.revision
+
+import org.oppia.app.model.Subtopic
+
+/** Listener for when a subtopic is selected for review. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revision/RevisionSubtopicSelector.kt]: Check this.

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt [hunk 5]
```diff
@@ -10,8 +10,8 @@ import org.oppia.app.model.Subtopic
 import javax.inject.Inject
 
 /** Fragment that card for topic review. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Check this

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardActivity.kt [hunk 6]
```diff
@@ -10,20 +10,20 @@ const val TOPIC_ID_ARGUMENT_KEY = "TOPIC_ID_"
 const val SUBTOPIC_ID_ARGUMENT_KEY = "SUBTOPIC_ID"
 
 /** Activity for review card. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardActivity.kt]: Check this

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardFragment.kt [hunk 7]
```diff
@@ -9,9 +9,9 @@ import org.oppia.app.fragment.InjectableDialogFragment
 import javax.inject.Inject
 
 /* Fragment that displays review card */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardFragment.kt]: Check this

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.RevisionCard
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
+import org.oppia.util.parser.HtmlParser
+import org.oppia.util.parser.RevisionCardHtmlParserEntityType
+import javax.inject.Inject
+
+/** [ViewModel] for Revision card, providing rich text and worked examples */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt]: Nit update:
`/** [ViewModel] for revision card, providing rich text and worked examples. */`

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  private lateinit var subtopicId: String
+  private lateinit var binding: RevisionCardFragmentBinding
+
+  var subtopicTitle: String = ""
+
+  val explanationLiveData: LiveData<CharSequence> by lazy {
+    processExplanationLiveData()
+  }
+
+  /** Sets the value of subtopicId and binding. Must be called before setting ViewModel to binding */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt]: Add full stop at the end of second sentence.

File: app/src/main/res/layout/revision_card_activity.xml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
   xmlns:tools="http://schemas.android.com/tools"
-  android:id="@+id/review_card_fragment_placeholder"
+  android:id="@+id/revision_card_fragment_placeholder"
   android:layout_width="match_parent"
   android:layout_height="match_parent"
-  tools:context="ReviewCardActivity" />
+  tools:context="Re" />
```
Comment (rt4914) [inline app/src/main/res/layout/revision_card_activity.xml]: Check this you are missing something here.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Changes suggested.
Also, make change in this file also https://github.com/oppia/oppia-android/blob/d6e58b45cc176170afa91d251e6d88b820a44a16/model/src/main/proto/topic.proto#L312

Also, please run the app and check if it is working or not because in current PR, I was not able to run this.

Also, always assign the PR for review, it was assigned to you, so because of which it got missed.

https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

File: app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt (hunk 1 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt]: Done

File: app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt (hunk 2 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/RouteToRevisionCardListener.kt]: Done

File: app/src/main/java/org/oppia/app/topic/TopicActivity.kt (hunk 3 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/TopicActivity.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revision/RevisionSubtopicSelector.kt (hunk 4 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revision/RevisionSubtopicSelector.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt (hunk 5 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardActivity.kt (hunk 6 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardActivity.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardFragment.kt (hunk 7 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardFragment.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt (hunk 8 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt (hunk 9 above)
Comment (abhinavraj23) [reply app/src/main/java/org/oppia/app/topic/revisioncard/RevisionCardViewModel.kt]: Done

File: app/src/main/res/layout/revision_card_activity.xml (hunk 10 above)
Comment (abhinavraj23) [reply app/src/main/res/layout/revision_card_activity.xml]: Done

**Comment by abhinavraj23** [comment]
@rt4914 all changes done, I have even tested the changes in my mobile, it's working perfectly fine!

**Review by rt4914 — CHANGES_REQUESTED** [review]
I am not able see any change in the file that I had mentioned in this comment: https://github.com/oppia/oppia-android/pull/844#pullrequestreview-379132897

Also, I am not able run the app, because you haven't done the above change, so how did it worked on your laptop?
Screenshot 2020-03-23 at 1 43 40 PM

**Comment by abhinavraj23** [comment]
@rt4914 can you please check now, I have made that change, and it was working for me because I guess I have made changes in my build files, please check now, I think it's fine now and the PR is ready to be merged

File: model/src/main/proto/topic.proto [hunk 11]
```diff
@@ -308,19 +308,19 @@ message Subtopic {
...
+// Corresponds to a revision card that can be displayed for a specific subtopic.
+message RevisionCard {
   // The title corresponding to the subtopic.
   string subtopic_title = 1;
 
-  // The core explanation of the subtopic being reviewed.
+  // The core explanation of the subtopic being revised.
   SubtitledHtml page_contents = 3;
 
   // Mapping from content_id to a VoiceoverMapping for each SubtitledHtml in this review card that has corresponding
```
Comment (rt4914) [inline model/src/main/proto/topic.proto]: Update `review` to `revision` in this.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@abhinavraj23  Even I think this should be the last one :-)

File: model/src/main/proto/topic.proto (hunk 11 above)
Comment (abhinavraj23) [reply model/src/main/proto/topic.proto]: Oops, just missed it, have done it now 👍

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**